pkg/cephfs/volumeoptions.go

```diff
@@ -40,7 +42,9 @@ func validateNonEmptyField(field, fieldName string) error {
 
 func (o *volumeOptions) validate() error {
 	if err := validateNonEmptyField(o.Monitors, "monitors"); err != nil {
-		return err
+		if err = validateNonEmptyField(o.MonValueFromSecret, "monValueFromSecret"); err != nil {
```

Comment:
What if both `monitors` and `monValueFromSecret` are non-empty?

https://github.com/ceph/ceph-csi/blob/3f196b5d739f0ed556f53a3e58b48850acbb9816/pkg/cephfs/volumeoptions.go#L103-L107

From the code, it's obvious that `monitors` would take precedence but the documentation doesn't mention that.

Reply:
So IMHO either:
1. a simple check that those two options can't be specified at the same time
2. add a comment in the docs that in case both of them are supplied, `monitors` will take precedence over `monValueFromSecret`

Reply:
good points, i updated it. Now when monitors are set in the secret, controller and node server favor monValueFromSecret. 

The rationale is that `monitors`, once set in storage class, are immutable. When monitors changes in the ceph cluster, existing PVs are no longer usable. In such case, a mutable setting in secret can solve this problem.